src/main/java/com/vg/apng/APNGWriter.java

```diff
@@ -42,29 +45,76 @@
             0x60, (byte) 0x82 //ae4260820
     };
 
-    public void write(Gray[] grays, File file) throws IOException {
+    /**
+     * Write an APNG image to a file.
+     * @param grays the grayscale images to write
+     * @param file the File to write to
+     * @param loopCount the number of time to loop the animation (0 means infinite)
+     * @throws IOException if the specified File is invalid
+     */
+    public void write(Gray[] grays, File file, int loopCount) throws IOException {
+        write(grays, new FileOutputStream(file), loopCount);
+    }
+
+    /**
+     * Write an APNG image to a FileOutputStream.
+     * @param grays the grayscale images to write
+     * @param os the OutputStream to write to
+     * @param loopCount the number of time to loop the animation (0 means infinite)
+     * @throws IOException if the specified OutputStream is invalid
+     */
+    public void write(Gray[] grays, OutputStream os, int loopCount) throws IOException {
         if (grays.length <= 0) {
             throw new RuntimeException("grays[] is empty");
         }
 
-        WritableByteChannel out = new FileOutputStream(file).getChannel();
-        try {
+        try (WritableByteChannel out = Channels.newChannel(os)) {
             out.write(ByteBuffer.wrap(PNG_SIG));
-            out.write(makeIHDRChunk(grays[0].width, grays[0].height));
-            out.write(make_acTLChunk(grays.length));
+            out.write(makeIHDRChunk(grays[0].getWidth(), grays[0].getHeight()));
```

Comment:
no need to create getter for a final primitive field (`grays[0].width` is equivalent to `grays[0].getWidth()`)